rust/private/rust.bzl

```
@@ -490,6 +490,32 @@ def _rust_test_impl(ctx):
 
     return providers
 
+def _rust_crate_group_impl(ctx):
+    crate_infos = []
+    dep_infos = []
+    runfiles = []
+
+    for dep in ctx.attr.deps:
+        if rust_common.crate_info in dep:
+            crate_infos.append(dep[rust_common.crate_info])
+            dep_infos.append(dep[rust_common.dep_info])
+        elif rust_common.crate_group_info in dep:
+            crate_infos.extend(dep[rust_common.crate_group_info].crate_infos)
+            dep_infos.extend(dep[rust_common.crate_group_info].dep_infos)
+        else:
+            fail("crate_group_info targets can only depend on rust_library or rust_crate_group targets.")
+
+        if dep[DefaultInfo].default_runfiles != None:
+            runfiles.append(dep[DefaultInfo].default_runfiles)
+
+    return [
+        rust_common.crate_group_info(
+            crate_infos = crate_infos,
+            dep_infos = dep_infos,
+        ),
+        DefaultInfo(runfiles = ctx.runfiles().merge_all(runfiles)),
```

For coverage support
```suggestion
        DefaultInfo(runfiles = ctx.runfiles().merge_all(runfiles)),
        coverage_common.instrumented_files_info(
            ctx,
            dependency_attributes = ["deps"],
        ),
```

---

ok. I'm not sure how to verify that this works, but I added these lines

---

I've been working on coverage, I can add a test for this in a future PR.